cli/stork-publisher-agent/model.go

```diff
@@ -0,0 +1,120 @@
+package stork_publisher_agent
+
+import "math/big"
+
+type MessageType string
+
+const WildcardSubscriptionAsset = "*"
+
+type ConnectionId string
+type AuthToken string
+
+type WebsocketMessage[T any] struct {
+	connId  ConnectionId
+	Type    string `json:"type"`
+	Error   string `json:"error,omitempty"`
+	TraceId string `json:"trace_id,omitempty"`
+	Data    T      `json:"data,omitempty"`
+}
+
+type ErrorMessage struct {
+	Error string `json:"error"`
+}
+
+type (
+	OracleId          string
+	AssetId           string
+	EvmPublisherKey   string
+	EvmPrivateKey     string
+	StarkPublisherKey string
+	StarkPrivateKey   string
+	PublisherKey      string
+	SignatureType     string
+	QuantizedPrice    string
+)
+
+const EvmSignatureType = SignatureType("evm")
+const StarkSignatureType = SignatureType("stark")
+
+// Incoming
+type PriceUpdatePullWebsocket struct {
+	PublishTimestamp int64   `json:"t"`
+	Asset            AssetId `json:"a"`
+	Price            float64 `json:"p"`
+}
+
+type ValueUpdatePushWebsocket struct {
+	PublishTimestamp int64   `json:"t"`
+	Asset            AssetId `json:"a"`
+	ValueString      string  `json:"vs"`
+	ValueFloat       float64 `json:"vf"`
```

Comment:
we now allow users to pass either a string value or a float value - the string will have no precision loss but could be a little slower, while the float will be fast with a small amount of precision loss. A string that doesn't actually parse to a float will send an error response via the websocket.